assembly/src/parsers/u32_ops.rs

```diff
@@ -181,22 +181,73 @@ pub fn parse_u32split(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), A
 // ARITHMETIC OPERATIONS
 // ================================================================================================
 
-/// Translates u32add assembly instruction to VM operation U32ADD.
+/// Translates u32add assembly instruction to VM operations.
+/// Specifically handles these specific inputs per the spec.
+/// - Zero argument: Assert the top two elemenets are u32 and push the result after ADD to the stack
+/// - Single argument:
+///   - "unsafe" skips the assert check and direclty performs a U32ADD EQZ NOT
+///   - "full" checks both numbers are u32 and perform the same operations as "unsafe"
+///   - Any number argument gets pushed to the stack, checked if both are u32 and performs a add.
 pub fn parse_u32add(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
     match op.num_parts() {
         0 => return Err(AssemblyError::missing_param(op)),
-        1 => span_ops.push(Operation::U32add),
+        1 => {
+            assert_u32(span_ops);
+            span_ops.push(Operation::Swap);
+            assert_u32(span_ops);
+            span_ops.push(Operation::Swap);
+            span_ops.push(Operation::Add);
```

Comment:
I don't think we need the `swap` here (since for addition the order doesn't matter). We also should use `u32add` instead of just `add`.